docs/integrations/piri-with-curio.md

```diff
@@ -0,0 +1,162 @@
+# Integrating Piri UCAN Server with Curio
+
+This guide helps existing Curio operators add Storacha network support by running a Piri UCAN server that uses Curio as the PDP backend.
+
+## Overview
+
+This integration allows you to:
+- Keep your existing Curio infrastructure
+- Add Storacha network compatibility
+- Accept uploads from Storacha clients
+- Leverage Curio's proven storage and proof capabilities
+
+Architecture:
+```
+Storacha Clients → Piri UCAN Server → Curio → Filecoin Network
+                  (This guide)      (Existing)
+```
+
+## Prerequisites
+
+Before starting, ensure you have:
+
+1. ✅ **Running Curio instance** with:
+   - HTTPS API endpoint (e.g., `https://curio.example.com`)
+   - Available storage capacity
+   - Active proof generation
+
+2. ✅ **System ready for Piri**:
+   - [Prerequisites met](../common/prerequisites.md)
+   - [Piri installed](../common/piri-installation.md)
+   - [PEM file generated](../common/pem-file-generation.md)
+
+3. ✅ **Network requirements**:
+   - Domain for UCAN server
+   - [TLS termination configured](../common/tls-termination.md)
+
+## Step 1: Configure Curio for Storacha
+
+### Add Storacha Service to Curio
+
+Curio needs to recognize the Piri UCAN server as an authorized service.
+
+1. **Extract your public key**:
+   ```bash
+   # From your service.pem file
+   openssl pkey -in service.pem -pubout
+   ```
+
+   Output example:
+   ```
+   -----BEGIN PUBLIC KEY-----
+   MCowBQYDK2VwAyEAYRpJI+aZVmKSFqYyVhMcOOeKYqBisNpReAGj4qF8kEg=
+   -----END PUBLIC KEY-----
+   ```
+
+2. **Configure Curio** to accept this key:
+   
+   Add the public key to Curio's configuration as a service named `storacha`. The exact method depends on your Curio version.
```

Comment:
Would be good to have an example here...

Reply:
https://github.com/storacha/piri/issues/114